pom.xml

```diff
@@ -1031,6 +1054,42 @@
         </plugins>
       </build>
     </profile>
+    <!-- Choose which hbase compatibility module we compile, test, and shade against.
+    Shouldn't have an effect on the generated code and artifacts apart from the contents of the the shaded jars
+    When specifying a non-default compatibility module, you also have to specify a
+    compatible hbase.version i.e mvn clean install -P phoenix-hbase-compat-2.2 -Dhbase.version=2.2.3
```

Comment:
Instead of forcing the user to activate the module and install it locally, you can simply add the appropriate compat module to the build reactor (specifying it with a new `<modules><module/></modules` element).

Yes, it forces us to rebuild the compat, but they both look very small at this point. The trick will be to make sure the appropriate one shows up in the assembly, not both of them (might automatically happen if all dependencies are getting pulled in properly).

Reply:
It already works that way, we always build and install both modules (maven artifacts)

However, I haven't thought of the /lib dir in the assembly, and the compat dependency doesn't get added automatically. I have fixed that now.